src/validation.cpp

```diff
@@ -1074,9 +1179,114 @@ PackageMempoolAcceptResult MemPoolAccept::AcceptMultipleTransactions(const std::
         }
     }
 
+    if (args.m_test_accept) return PackageMempoolAcceptResult(package_state, std::move(results));
+
+    if (!FinalizePackage(args, workspaces, package_state, results)) {
+        package_state.Invalid(PackageValidationResult::PCKG_TX, "submission failed");
```

Comment:
"submission failed" doesn't add any new information. I think `FinalizePackage()` can only fail due to:

- logic error
- mempool full (package partially accepted or not accepted)

Can we distinguish those and return the information in the `PackageValidationState` object?

Reply:
Done in #23804